pkg/contexts/ocm/transfer/transferhandler/transferhandler.go

```diff
@@ -11,12 +11,65 @@ import (
 	"github.com/open-component-model/ocm/pkg/errors"
 )
 
-type TransferOptions interface{}
+const KIND_TRANSFEROPTION = "transfer option"
 
+// TransferOptions is the general type for an option bag
+// holding options for a transfer handler.
+// Different transfer handler implementations may use differ
+// concrete option sets.
+// Therefore, every option set uses its own options type
+// The options setters/getters MUST be interfaces, which may be implemented
+// by different option set types to enable the option implementations
+// to work with different options sets.
+type TransferOptions interface {
+	NewTransferHandler() (TransferHandler, error)
+}
+
+type TransferOptionsCreator interface {
+	NewOptions() TransferOptions
+}
+
+// TransferOption is an option used to configure a transfer handler.
+// Different transfer handlers may use different sets of transfer options
+// by an own implementation of the TransferOptions interface enriched
+// by own supported option interfaces.
+// Options are gathered in such objects acting as option bag.
+// To enable options to be used with different such implementations
+// every option MUST have own interfaces for the concrete option getter/setter,
+// which are used by the options to apply themselves to an option bag.
+// This way different implementation of TransferOptions can share options
+// configurable by the same option functions.
+//
+// For example the spiff transfer handler options include all the standard
+// handler options.
 type TransferOption interface {
+	TransferOptionsCreator
 	ApplyTransferOption(TransferOptions) error
 }
 
+type optionsPointer[P any] interface {
+	TransferOptions
+	*P
+}
+
+// SpecilizedOptionsCreator is the base implementation for options objects
+// for specialized transfer handlers.
+type SpecilizedOptionsCreator[P optionsPointer[T], T any] struct{}
+
+func (o SpecilizedOptionsCreator[P, T]) NewOptions() TransferOptions {
+	var opts T
+	return P(&opts)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+// TransferHandler controls the transfer of component versions.
+// It can be used to control the value transport of sources and resources
+// on artifact level and the way how nested component version are transported.
```

Comment:
>on artifact level

What is on artifact level? What's an artifact in this context?

Reply:
Any decision about the process can be based of attributes of the described artifact (common term for resource and source, see ocm-spec).